### PR TITLE
feat(other): fake e-mail server for end-to-end testing

### LINKED ISSUE #1257: 🚀 [Feature] Fake Email Server
## 🚀 Feature

Implement a fake Email server to test emails locally.
See Ocelot for how to do that

### PR DESCRIPTION
**Pull request by mahula**
## 🍰 Pullrequest

Set up a fake e-mail server ([maildev](https://maildev.github.io/maildev/)) for proper e-mail testing locally while developing and in the Github test workflow.

### Issues
- fixes #1257

### CODE REVIEW

File: backend/src/config/index.ts [hunk 1]
```diff
@@ -66,14 +66,16 @@ const loginServer = {
 }
 
 const email = {
-  EMAIL: process.env.EMAIL === 'true' || false,
+  // eslint-disable-next-line no-unneeded-ternary
+  EMAIL: process.env.EMAIL === 'false' ? false : process.env.NODE_ENV !== 'development',
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply backend/src/config/index.ts:70]: I agree, we might need to refactor this email config mess before we do hacks like this

File: backend/src/config/index.ts [hunk 2]
```diff
@@ -66,14 +66,16 @@ const loginServer = {
...
   EMAIL_TEST_MODUS: process.env.EMAIL_TEST_MODUS === 'true' || false,
   EMAIL_TEST_RECEIVER: process.env.EMAIL_TEST_RECEIVER || 'stage1@gradido.net',
-  EMAIL_USERNAME: process.env.EMAIL_USERNAME || 'gradido_email',
+  EMAIL_USERNAME: process.env.EMAIL_USERNAME || 'null',
   EMAIL_SENDER: process.env.EMAIL_SENDER || 'info@gradido.net',
-  EMAIL_PASSWORD: process.env.EMAIL_PASSWORD || 'xxx',
-  EMAIL_SMTP_URL: process.env.EMAIL_SMTP_URL || 'gmail.com',
-  EMAIL_SMTP_PORT: process.env.EMAIL_SMTP_PORT || '587',
+  EMAIL_PASSWORD: process.env.EMAIL_PASSWORD || 'null',
+  EMAIL_SMTP_URL: process.env.EMAIL_SMTP_URL || 'mailserver',
```
Comment (ulfgebhardt) [inline backend/src/config/index.ts:76]: localhost for non-docker things? Does it work both in docker and non-docker?

**Review by ulfgebhardt — COMMENTED** [review]
Looks great already - i think this is almost there. Lets have a discussion about the email config mess

File: backend/src/config/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply backend/src/config/index.ts:70]: We agreed that the default will be false, not true (current implementation) and that the developer has to set this manually to true in order to have this feature enabled

[the remark this answers is not in the document]
Comment (mahula) [reply backend/src/config/index.ts]: This has been fixed with https://github.com/gradido/gradido/commit/881e38f22b168a0533d61f211b63dcc2c51a8260 setting email disabled as default as discussed.

File: backend/src/config/index.ts [hunk 3]
```diff
@@ -66,14 +66,16 @@ const loginServer = {
...
+  EMAIL_SMTP_PORT: process.env.EMAIL_SMTP_PORT || '1025',
+  EMAIL_TLS: process.env.EMAIL_TLS === 'true' || false,
```
[the remark this answers is not in the document]
Comment (mahula) [reply backend/src/config/index.ts]: No, probably not.
Fixed with https://github.com/gradido/gradido/commit/881e38f22b168a0533d61f211b63dcc2c51a8260

File: backend/src/config/index.ts [hunk 4]
```diff
@@ -69,11 +69,13 @@ const email = {
...
   EMAIL_TEST_RECEIVER: process.env.EMAIL_TEST_RECEIVER || 'stage1@gradido.net',
-  EMAIL_USERNAME: process.env.EMAIL_USERNAME || 'gradido_email',
+  EMAIL_USERNAME: process.env.EMAIL_USERNAME || 'null',
   EMAIL_SENDER: process.env.EMAIL_SENDER || 'info@gradido.net',
-  EMAIL_PASSWORD: process.env.EMAIL_PASSWORD || 'xxx',
-  EMAIL_SMTP_URL: process.env.EMAIL_SMTP_URL || 'gmail.com',
-  EMAIL_SMTP_PORT: process.env.EMAIL_SMTP_PORT || '587',
+  EMAIL_PASSWORD: process.env.EMAIL_PASSWORD || 'null',
+  EMAIL_SMTP_URL: process.env.EMAIL_SMTP_URL || 'mailserver',
+  EMAIL_SMTP_PORT: process.env.EMAIL_SMTP_PORT || '1025',
```
Comment (ulfgebhardt) [inline backend/src/config/index.ts:76]: Can we configure maildev to run on std port `587`?

File: backend/src/config/index.ts [hunk 5]
```diff
@@ -69,11 +69,13 @@ const email = {
   EMAIL: process.env.EMAIL === 'true' || false,
   EMAIL_TEST_MODUS: process.env.EMAIL_TEST_MODUS === 'true' || false,
   EMAIL_TEST_RECEIVER: process.env.EMAIL_TEST_RECEIVER || 'stage1@gradido.net',
-  EMAIL_USERNAME: process.env.EMAIL_USERNAME || 'gradido_email',
+  EMAIL_USERNAME: process.env.EMAIL_USERNAME || 'null',
```
Comment (ulfgebhardt) [inline backend/src/config/index.ts:72]: please no string

File: backend/src/config/index.ts [hunk 6]
```diff
@@ -69,11 +69,13 @@ const email = {
...
   EMAIL_SENDER: process.env.EMAIL_SENDER || 'info@gradido.net',
-  EMAIL_PASSWORD: process.env.EMAIL_PASSWORD || 'xxx',
-  EMAIL_SMTP_URL: process.env.EMAIL_SMTP_URL || 'gmail.com',
-  EMAIL_SMTP_PORT: process.env.EMAIL_SMTP_PORT || '587',
+  EMAIL_PASSWORD: process.env.EMAIL_PASSWORD || 'null',
```
Comment (ulfgebhardt) [inline backend/src/config/index.ts:74]: please no string

File: docker-compose.override.yml [hunk 7]
```diff
@@ -112,6 +112,17 @@ services:
...
      - /sessions
 
+  ########################################################
+  # MAILSERVER TO FAKE SMTP ##############################
+  ########################################################
+  mailserver:
+    image: maildev/maildev
+    ports:
+      - 1080:1080
+      - 1025:1025
```
Comment (ulfgebhardt) [inline docker-compose.override.yml:122]:
```suggestion
      - 587:1025
```

File: backend/src/config/index.ts (hunk 5 above)
Comment (mahula) [reply backend/src/config/index.ts]: This is fixed in https://github.com/gradido/gradido/commit/1e265b1078c10cfb2a7aeaebeaecbca4e6983780.

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**